### ISSUE TITLE
1.37.0: `test_codec_built_in_equivalent` fails with pytest

### ISSUE THREAD

**Issue by mtelka** [adrienverge/yamllint#734]
When I run `yamllint` tests using `pytest` then the `test_codec_built_in_equivalent` test fails:
```
_______________ ERROR at setup of test_codec_built_in_equivalent _______________
file /data/builds/ul-mass-rebuild/components/python/yamllint/build/amd64-3.9/tests/common.py, line 107
  def test_codec_built_in_equivalent(test_codec):
E       fixture 'test_codec' not found
>       available fixtures: _xunit_setup_module_fixture_tests.test_decoder, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, monkeypatch, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/data/builds/ul-mass-rebuild/components/python/yamllint/build/amd64-3.9/tests/common.py:107
```
When I switch to `unittest` then all tests pass.

This is minor issue, but it would be great if `pytest` works OOTB too.

**Comment by mtelka** [comment]
This is regression from version 1.36.2 where pytest worked properly.

**Comment by adrienverge** [comment]
Hello,

I confirm I can reproduce the same error with:

    pytest tests/test_decoder.py

I don't understand the problem, because `test_codec_built_in_equivalent()` is just pure Python:

```python
class EncodingStuffFromCommonTestCase(unittest.TestCase):
    def test_test_codec_built_in_equivalent(self):
        self.assertEqual(
            'utf_32',
            test_codec_built_in_equivalent('utf_32_be_sig')
        )

def test_codec_built_in_equivalent(test_codec):
    return_value = test_codec
    for suffix in ('_sig', '_be', '_le'):
        return_value = return_value.replace(suffix, '')
    return return_value
```

Has `test_codec` a special meaning with pytest?

@Jayman2000 you wrote this code (added in commit a53fa80 "decoder: Autodetect encoding of most YAML files"): maybe you have an idea?

**Comment by Jayman2000** [comment]
> [@Jayman2000](https://github.com/Jayman2000) you wrote this code (added in commit [a53fa80](https://github.com/adrienverge/yamllint/commit/a53fa80e80e3de3eb329609fdf270bb24f7e6e81) "decoder: Autodetect encoding of most YAML files"): maybe you have an idea?

I think that I figured out why this failure is happening. I wrote about why I think that it’s happening in the commit message for #736.
